pkg/operator/kube_cloud_config/transform.go

```diff
@@ -0,0 +1,99 @@
+package kube_cloud_config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorclient "github.com/openshift/cluster-config-operator/pkg/operator/operatorclient"
+)
+
+func CloudConfigTransformers() map[configv1.PlatformType]cloudConfigTransformer {
+	cloudConfigTransformers := map[configv1.PlatformType]cloudConfigTransformer{
+		configv1.AWSPlatformType:   awsTransformer,
+		configv1.AzurePlatformType: azureTransformer,
+	}
+
+	return cloudConfigTransformers
+}
+
+// ValidateFile verifies a file exists, has content, and is a regular file
+func ValidateFile(path string) error {
+	st, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if !st.Mode().IsRegular() {
+		return fmt.Errorf("%s is not a regular file.", path)
+	}
+	if st.Size() <= 0 {
+		return fmt.Errorf("%s is empty.", path)
+	}
+	return nil
+}
+
+func BootstrapTransformer(infrastructureFile string, cloudProviderFile string) ([]byte, error) {
+
+	// Read, parse, and save the infrastructure object
+	var clusterInfrastructure configv1.Infrastructure
+	fileData, err := ioutil.ReadFile(infrastructureFile)
+	if err != nil {
+		return nil, err
+	}
+	if len(fileData) > 0 {
```

Comment:
Is this check required?

Reply:
Apparently not. I did not realize it's safe to unmarshal a zero length bytes object.